api/integrations/launch_darkly/services.py

```diff
@@ -165,6 +175,87 @@ def _create_string_feature_states(
         )
 
 
+# Based on: https://docs.launchdarkly.com/sdk/concepts/flag-evaluation-rules#operators
+def _ld_operator_to_flagsmith_operator(ld_operator: str) -> Optional[str]:
+    return dict(
+        {
+            "in": constants.IN,
+            "endsWith": constants.REGEX,
+            "startsWith": constants.REGEX,
+            "matches": constants.REGEX,
+            "contains": constants.CONTAINS,
+            "lessThan": constants.LESS_THAN,
+            "lessThanOrEqual": constants.LESS_THAN_INCLUSIVE,
+            "greaterThan": constants.GREATER_THAN,
+            "greaterThanOrEqual": constants.GREATER_THAN_INCLUSIVE,
+            "before": constants.LESS_THAN,
+            "after": constants.GREATER_THAN,
+            "semVerEqual": constants.EQUAL,
+            "semVerLessThan": constants.LESS_THAN,
+            "semVerGreaterThan": constants.GREATER_THAN,
+            "segmentMatch": constants.REGEX,
+        }
+    ).get(ld_operator, None)
+
+
+# TODO: Not sure what happens if it is not `IN` and there are multiple values.
+def _convert_ld_value(values: list[str], ld_operator: str) -> str:
+    match ld_operator:
+        case "in":
+            return ",".join(values)
+        case "endsWith":
+            return ".*" + re.escape(values[0])
+        case "startsWith":
+            return re.escape(values[0]) + ".*"
+        case "matches" | "segmentMatch":
+            return re.escape(values[0]).replace("\\*", ".*")
+        case "contains":
+            return ".*" + re.escape(values[0]) + ".*"
+        case _:
+            return values[0]
+
+
+def _clauses_to_segment(
+    clauses: list[Clause],
+    project: Project,
+    feature: Feature,
+    environment: Environment,
+    name: str,
+) -> Segment:
+    segment = Segment.objects.create(name=name, project=project, feature=feature)
+
+    parent_rule = SegmentRule.objects.create(segment=segment, type=SegmentRule.ANY_RULE)
+
+    child_rule = SegmentRule.objects.create(rule=parent_rule, type=SegmentRule.ANY_RULE)
+
+    for clause in clauses:
+        operator = _ld_operator_to_flagsmith_operator(clause["op"])
+        value = _convert_ld_value(clause["values"], clause["op"])
+        _property = clause["attribute"]
+
+        if operator is not None:
+            condition = Condition.objects.update_or_create(
+                rule=child_rule,
+                property=_property,
+                value=value,
+                operator=operator,
+                created_with_segment=True,
+            )
+            logger.warning("Condition created: " + str(condition))
+        else:
+            logger.warning(
+                "Can't map launch darkly operator: " + clause["op"] + ", skipping..."
+            )
```

Comment:
I think we need to use `_log_error` here to mark the import request as failed.

Reply:
We don't expect any other operators (based on SDK) but from User's perspective, isn't it better if we skip silently?

Both approaches just work fine, I believe the biggest blocker of skipping silently is how can we communicate this with the user 🙂

Reply:
As far as I understand, there is no `not in` kind of operators so we won't be able to import everything. Basically all clauses in launch darkly are negatable by a flag, couldn't find the equivalent in Flagsmith so we definetly have to sacrifice some rules here.

Reply:
I am going to leave this as a warning because we will need a better way to communicate which flags / rules are not imported successfully.

Can fix it all together with other stuff.